doc/devel/coding_guide.rst

```diff
@@ -165,6 +165,48 @@ C/C++ extensions
 
 
 
+PR Review guidelines
+====================
+
+* We run on a 'use your judgement' system (for better or worse) and do
+  not have a strict nose-count rule.  If you have a commit bit, then
+  you are trusted to use it and to know when to ask for more review.
+
+* Do not self merge with the exception of un-breaking travis.
+
+* Squashing is case-by-case.  The balance is between burden on the
+  contributor, keeping a relatively clean history, and keeping a
+  history usable for bisecting.  The only time we are really strict
+  about it is to eliminate binary files (ex multiple test image
+  re-generations) and to remove upstream merges.
+
+* Be patient with new contributors.
+
+
+Backports
+=========
+
+
+When doing back ports please include the branch you backported the
+commit to along with the SHA in a comment on the original PR.
+
+A quick guide to how to back port a merge commit::
+
+  git remote update
+  git checkout target_branch
+  git merge --ff-only matplotlib/target_branch
+  git cherry-pick -m 1 TARGET_SHA
+  gitk   # to look at it
+  # local tests? (use your judgment)
+  git push DANGER target_branch
+  # leave a comment on PR noting sha of the resulting commit
+  # from the cherry-pick + branch it was moved to
+
+here matplotlib is a read-only remote to the matplotlib/matplotlib repo and DANGER in a read/write remote to the matplotlib/matplotlib repo.
```

Comment:
This is a sub-section of my `git remote -v`

```
DANGER  git@github.com:matplotlib/matplotlib.git (fetch)
DANGER  git@github.com:matplotlib/matplotlib.git (push)
matplotlib      git://github.com/matplotlib/matplotlib.git (fetch)
matplotlib      git://github.com/matplotlib/matplotlib.git (push)
tacaswell       git@github.com:tacaswell/matplotlib.git (fetch)
tacaswell       git@github.com:tacaswell/matplotlib.git (push)
```

I also use ssh keys on machines where I can (I also have 2FA turned on so https requires an oauth token) so I can push to `tacaswel` and `DANGER`, but not `matplotlib`. 

```
$ git push matplotlib v2.x
fatal: remote error: 
  You can't push to git://github.com/matplotlib/matplotlib.git
  Use https://github.com/matplotlib/matplotlib.git
```

 Due to the fear of pushing to the wrong remote (which I have done) I am also very pedantic about being explicit about remotes when pushing and have git turned down to the least magical 'guess what you mean by `git push`'.  

Having to type the word 'DANGER' in all caps (or at least see it in magit) does remind you you are about to do something dangerous ;) 

I got this naming idea from @dmcdougall